purchase_requisition_bid_selection/tests/test_generate_po.py

```diff
@@ -0,0 +1,102 @@
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+import openerp.tests.common as common
+from openerp import fields
+
+
+class test_generate_po(common.TransactionCase):
+    """ Test genertation of a purchase order from selected Bid and ensure
+    unselected Bids are cancelled
+    """
+
+    def setUp(self):
+        super(test_generate_po, self).setUp()
+        PReq = self.env['purchase.requisition']
+        PO = self.env['purchase.order']
+        POL = self.env['purchase.order.line']
+
+        # Put purchase.requisition directly to 'closed'
+        dummy_wkf_trans = self.env['workflow.transition'].create({
+            'act_from': self.env.ref('purchase_requisition.act_draft').id,
+            'act_to': self.env.ref(
+                'purchase_requisition_bid_selection.act_closed').id,
+            'condition': True,
+        })
+        self.preq = PReq.create({'state': 'closed'})
+        dummy_wkf_trans.unlink()
+
+        partner_12 = self.env.ref('base.res_partner_12')
+        partner_13 = self.env.ref('base.res_partner_13')
+        partner_14 = self.env.ref('base.res_partner_14')
+
+        # Put purchase.oders directly to 'bid'
+        dummy_wkf_trans = self.env['workflow.transition'].create({
+            'act_from': self.env.ref('purchase.act_draft').id,
+            'act_to': self.env.ref('purchase.act_bid').id,
+            'condition': True,
+        })
+        self.bid_selected = PO.create(
+            {'state': 'bid',
+             'bid_partial': True,
+             'partner_id': partner_12.id,
+             'location_id': self.env.ref('stock.stock_location_stock').id,
+             'pricelist_id': partner_12.property_product_pricelist_purchase.id,
+             })
+        pol = POL.create({
+            'state': 'confirmed',
+            'order_id': self.bid_selected.id,
+            'name': '/',
+            'price_unit': 100,
+            'date_planned': fields.Datetime.now(),
+            })
+        self.bid = PO.create(
+            {'state': 'bid',
+             'partner_id': partner_13.id,
+             'location_id': self.env.ref('stock.stock_location_stock').id,
+             'pricelist_id': partner_13.property_product_pricelist_purchase.id,
+             })
+        dummy_wkf_trans.unlink()
+
+        # draftbid is at same workflow state as draft
+        self.draftbid = PO.create(
+            {'state': 'draftbid',
+             'partner_id': partner_14.id,
+             'location_id': self.env.ref('stock.stock_location_stock').id,
+             'pricelist_id': partner_14.property_product_pricelist_purchase.id,
+             })
+
+        purchases = self.env['purchase.order']
+        purchases |= self.bid_selected
```

Comment:
this is perfectly fine for me, but you can also say
`purchases = self.bid_selected | self.bid | self.draftbid`